lhotse/dataset/speech_recognition.py

```diff
@@ -60,6 +62,172 @@ def __len__(self) -> int:
         return len(self.cut_ids)
 
 
+class K2SpeechRecognitionIterableDataset(IterableDataset):
```

Comment:
I think i would be clearer to inherit from `torch.IterableDataset` explicitly, so readers will know it's not a k2 thing.